server/providers/combined/tools/write-next-story/shell-story-parser.ts

```diff
@@ -19,12 +23,460 @@ export interface ParsedShellStory {
   lowPriority: string[];   // ⏬ bullets
   excluded: string[];      // ❌ bullets
   questions: string[];     // ❓ bullets
-  rawContent: string;      // Full markdown for this story
+  rawContent: string;     // Original markdown for AI prompts
+}
+
+// ============================================================================
+// ADF-Based Parsing (Preferred - Preserves Formatting)
+// ============================================================================
+
+/**
+ * Parse shell stories from ADF bullet list structure
+ * @param shellStoriesSection - ADF nodes containing Shell Stories section (including heading)
+ * @returns Array of parsed shell stories
+ * 
+ * @example
+ * const { section } = extractAdfSection(epicDescription.content, "Shell Stories");
+ * const stories = parseShellStoriesFromAdf(section);
+ */
+export function parseShellStoriesFromAdf(
+  shellStoriesSection: ADFNode[]
+): ParsedShellStoryADF[] {
+  const stories: ParsedShellStoryADF[] = [];
+  
+  // Find bulletList nodes in section
+  for (const node of shellStoriesSection) {
+    if (node.type === 'bulletList' && node.content) {
+      // Each listItem is a shell story
+      for (const listItem of node.content) {
+        const story = parseShellStoryFromListItem(listItem);
+        if (story) {
+          stories.push(story);
+        }
+      }
+    }
+  }
+  
+  return stories;
+}
+
+/**
+ * Add completion marker to shell story in ADF
+ * 
+ * Adds a link mark to the title text node and appends a timestamp.
+ * Format: `st001` **[Title](https://url)** ⟩ Description _(2025-01-15T10:30:00Z)_
+ * 
+ * @param shellStoriesSection - Shell Stories ADF nodes (including heading)
+ * @param storyId - Story ID to mark (e.g., "st001")
+ * @param issueKey - Jira issue key (e.g., "PROJ-123")
+ * @param issueUrl - Jira issue URL
+ * @returns New section with marker added
+ * 
+ * @example
+ * const updated = addCompletionMarkerToStory(
+ *   shellStoriesSection,
+ *   "st001",
+ *   "PROJ-123",
+ *   "https://bitovi.atlassian.net/browse/PROJ-123"
+ * );
+ */
+export function addCompletionMarkerToStory(
```

Comment:
This should probably be `addCompletionMarkerToShellStory`. It's probably good to distinguish between "stories" and "shell stories".